federationapi/routing/send.go

```diff
@@ -482,14 +483,10 @@ func (t *txnReq) retrieveMissingAuthEvents(
 		missingAuthEvents[missingAuthEventID] = struct{}{}
 	}
 
-	servers := t.getServers(ctx, e.RoomID())
-	if len(servers) > 5 {
-		servers = servers[:5]
```

Comment:
For rooms like Matrix HQ this could result in slow message processing..

Reply:
Well, in the worst case when the first set of servers we try are not available, but conversely, in a P2P room, we can't assume servers will be up. It's probably better to just try more servers rather than give up and drop the event.